### DISCUSSION TITLE
Baseline and cached builds

### DISCUSSION THREAD

**Discussion by na-ji** [percy/cli#1252] [Support]
Hello,

I'm developing a bunch of apps in a NX monorepo with [NX distributed caching](https://nx.app/) enabled. With this cache enabled, it means that a test run won't be executed again if the files are still the same. Some of my apps are covered by Cypress jobs that also take Percy snapshots. 

However, those jobs can be cached, so it means that sometime no Percy job are made. This is a problem when a PR is merged, and the Cypress job is not executed again since it's cached. Because of that, no baseline is saved on Percy. Since I would like to keep my Cypress builds cached (they can take a very long time), I have two questions that could lead to a solution:
- Would it be possible to set a previous build as baseline? This could be used to set a PR build as baseline when merging.
- Would it be possible to delay the snapshots upload? With this, the snapshots could be saved in the Cypress job cache, and then a job would be run without cache to upload those snapshots, even if the Cypress job was cached previously. 

Thank you for your help.

**Comment by itsjwala** [comment]
Hey @na-ji,
please reach out to us via support https://www.browserstack.com/support/percy we'll have a look
